lib/test_transform.dart

```diff
@@ -33,43 +36,37 @@ class TestTransform implements Transform {
   Asset _primaryAsset;
   AssetId _primaryAssetId;
   bool consumed = false;
-  String packageRoot;
+  Uri dotPackages;
+  PackageResolver packageResolver;
 
   final List<MessageRecord> messages = <MessageRecord>[];
 
   logFunction(AssetId id, LogLevel level, String message, SourceSpan span) {
     messages.add(new MessageRecord(id, level, message, span));
   }
 
-  TestTransform(this.fileName, this.fileContents, this.package,
-      [String packageRoot]) {
-    this.packageRoot =
-        packageRoot == null ? io.Platform.packageRoot : packageRoot;
-    // The semantics of `io.Platform.packageRoot` suddenly changed from version
-    // dart-sdk-1.14.0-dev.4.0 to .5.0: It used to be passed on from the
-    // command line arguments directly, but starting from .5.0 it is being
-    // interpreted: If it is a relative path then it is turned into an absolute
-    // path, and an absolute path is in turn changed into a 'file://' uri.
-    // If [packageRoot] is a 'file://' uri then we reduce it to a path.
-    Uri uri;
-    if (this.packageRoot == null) {
-      logger.error("Cannot determine the package root, using './packages'.");
-      this.packageRoot = "./packages";
-    }
-    try {
-      uri = Uri.parse(this.packageRoot);
-      this.packageRoot = uri.path;
-    } on FormatException catch (_) {
-      // No problem, now we just know that `packageRoot` is not a well-formed
-      // uri; it should then be a path.
-    }
-    if (this.packageRoot.endsWith(io.Platform.pathSeparator)) {
-      this.packageRoot =
-          this.packageRoot.substring(0, this.packageRoot.length - 1);
-    }
+  TestTransform(
+      this.fileName, this.fileContents, this.package) {
     _primaryAssetId = new AssetId.parse(fileName);
     _primaryAsset = new Asset.fromString(_primaryAssetId, fileContents);
     assets[_primaryAssetId] = _primaryAsset;
+    Uri scriptUri = io.Platform.script;
+    String uriDataScriptString = scriptUri.data?.contentAsString();
+    if (uriDataScriptString != null) {
+      // `uriDataScriptString` actually contains the entire program, which
+      // imports the test file using a `file:///` uri.
+      int startIndex = uriDataScriptString.indexOf("file:///");
```

Comment:
Perhaps use a regexp including the whole line you want to match.

Reply:
Right, and that would only match a single line, so we wouldn't get the wrong result even if we were to receive this:
```dart
...
import 'file:///whatever/I-am-just-cheating-you.dart';
import 'file:///this/is/the/one/we/want/test_reflectable/test/some_test.dart';
...
```
Done.